src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsPackageInstaller.cs

```diff
@@ -53,7 +59,26 @@ public class VsPackageInstaller : IVsPackageInstaller2
 
             _projectContext = new Lazy<INuGetProjectContext>(() => new VSAPIProjectContext());
 
-            PumpingJTF = new PumpingJTF(NuGetUIThreadHelper.JoinableTaskFactory.Context);
+            PumpingJTF = null;
+        }
+
+        private void RunJTFWithCorrectContext(Project project, Func<Task> asyncTask)
+        {
+            if (PumpingJTF == null)
```

Comment:
You need to check for CPS capability for each project, these apis are being called. Since there could be a case where initial project was not CPS enabled but later on one more project was added which was CPS. But in your case, you only check first time.